pkg/controllers/state/cluster.go

```diff
@@ -464,6 +476,16 @@ func (c *Cluster) updatePod(ctx context.Context, pod *v1.Pod) error {
 		err = c.updateNodeUsageFromPod(ctx, pod)
 	}
 	c.updatePodAntiAffinities(pod)
+	if pod.DeletionTimestamp != nil {
+		podDeletionTime := pod.DeletionTimestamp.UnixMilli()
+		if podDeletionTime > atomic.LoadInt64(&c.lastPodDeletionTime) {
```

Comment:
Don't these comparisons create races where two loads happen between the store call on 482?

Reply:
A way to resolve this might be to use the `atomic.CompareAndSwap...` call

Reply:
I don't think it matters. The deletion timestamp has a one-second resolution IIRC and if something else stores with a different value, it won't be off by more than a second.  Changing the code to do compare and swap makes it even more complicated:

```go
	if pod.DeletionTimestamp != nil {
		podDeletionTime := pod.DeletionTimestamp.UnixMilli()
		for {
			lastPodDeletionTime := atomic.LoadInt64(&c.lastPodDeletionTime)
			if (podDeletionTime > lastPodDeletionTime && atomic.CompareAndSwapInt64(&c.lastPodDeletionTime, lastPodDeletionTime, podDeletionTime)) ||
				podDeletionTime <= lastPodDeletionTime {
				break
			}
		}
	}
```

Reply:
Makes sense. Yeah, I guess in this case, the race condition is warranted. Can we make a comment to note that a race could happen in these two locations?

My main concern is someone coming in the future and assuming that this code section is synchronous without being aware of this possible race.